ADR/0023-git-references-to-furnish-integration-test-scenarios.md

```diff
@@ -0,0 +1,72 @@
+# 23. Git references to furnish Integration Test Scenarios
+
+* Date Documented: 2023-04-10
+* Date Accepted: TBD
+
+## Status
+
+In Review
+
+## Context
+
+Up to now, the Integration service has only supported setting the Tekton bundle (Bundle string `json:"bundle"`) in the 
+IntegrationTestScenario [CR] as a reference for the integration tests, in order to run the Tekton PipelineRuns.
+
+Below is the existing format to supply the Integration test references:
+
+![](../diagrams/ADR-0023/tekton-bundle-ITS.jpg)
+
+Building and maintaining Tekton bundles is cumbersome for users when providing custom tests.
+
+Resolving tasks from a git repository is more natural for developers and allows for faster iteration. Hence, users
+should be able to reference their Tekton pipeline definitions that are hosted in git repositories ([RHTAP-402]).
+
+To support the functionality of using git references, Tekton resolvers need to be enabled. As part of [PLNSRVCE-1030], 
+pipeline service adds support for the [Tekton-resolvers] including git resolvers which should in turn give us the ability
+to expand the support for resolvers in the IntegrationTestScenarios.
+
+## Decision
+
+Integration service now migrates to the new version of the IntegrationTestScenario CRD having the expanded functionality
+to allow using different types of Tekton resolvers ex: bundles-resolver, cluster-resolver, hub-resolver along with 
+the git-resolver. This gives the extensibility to permit the use of different resolvers in the future.
+
+The new format of adding the Integration Test Scenarios would look like:
+
+
+![](../diagrams/ADR-0023/git-references-ITS.jpg)
+
+Users git repository path to their test code can now provide the information or the location of their IntegrationTestScenarios
+through Git references like **Github URL**, **Git Options** (**Revision** & **Path in repository**). 
+
+The information provided by the users in the above form will then be consumed by the integration service with the help of IntegrationTestScenario CR and will be used to run the integration tests against the Application Snapshot in question.
+
+Example of the tekton pipeline definition with git resolver:
+
+![](../diagrams/ADR-0023/tekton-pipeline-definition-git-resolver.jpg)
+
+
+
+## Consequences
+
+* Better user experience through Tekton pipeline definitions that are hosted in git repositories which are easier to 
+maintain as opposed to build and maintain the Tekton bundle images.
```

Comment:
```suggestion
maintain as opposed to building and maintaining the Tekton bundle images.
```

Reply:
```suggestion
maintain as opposed to building their own Tekton bundle images every time they adjust their test. It also removes the need for our users to have to understand what a Tekton bundle is and learn how to build it, if they don't already know.
```

What you had was fair, but I think undersells the improvement this provides for our users. Not sure how in depth the ADR needs to go though. I also dropped the "and maintaining" in my suggestion since it already mentioned that it was easier to maintain earlier in the sentence.

Reply:
ack